server/src/main/java/org/elasticsearch/action/ingest/SimulateExecutionService.java

```diff
@@ -69,6 +69,12 @@ public void execute(SimulatePipelineRequest.Parsed request, ActionListener<Simul
             final AtomicInteger counter = new AtomicInteger();
             final List<SimulateDocumentResult> responses =
                 new CopyOnWriteArrayList<>(new SimulateDocumentBaseResult[request.getDocuments().size()]);
+
+            if (request.getDocuments().isEmpty()) {
+                l.onResponse(new SimulatePipelineResponse(request.getPipeline().getId(),
```

Comment:
Also maybe add a return statement here? Technically it is not needed because the for loop is skipped, but I find it easier to read.